docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -505,988 +1780,2081 @@ Use a ``--config`` file to define multiple options or reflect an infrastructure-
-``Alice_1``, ``Alice_2``, and ``Peter-1``.
+Commonly used configuration arguments include:
 
 More advanced expressions can make use of brackets, such as
 ``--pipeline-parties="Alice* | Bob* | (participant* & !(participant3::*))"``.
 
-
 Handle Configuration Changes
-============================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-PQS initializes its behavior on startup by reading its configuration files.
-It currently doesn't support dynamic configuration updates, so making a
+PQS initializes its behavior on startup by reading its configuration
+files. It does not perform dynamic configuration updates, so making a
 configuration change (such as adding a new party, new template, or new
-interface) requires stopping PQS, modifying its configuration, and then
-starting PQS. On startup, PQS will read the updated configuration.
+interface) requires a restart.
+
+When the configuration changes, the default is that PQS will not go back
+in time (older offset) but only move forward in time (current watermark
+offset and newer). If the database is dropped, then PQS can be started
+at the oldest, unpruned offset of the participant node and use the
+Participant node history to extract the events based on the updated
+configuration.
 
-When the configuration changes, the default is that PQS will not go back in
-time (older offset) but only move forward in time (current watermark offset
-and newer). If the database is dropped, then PQS can be started at the
-oldest, unpruned offset of the participant node and use the participant node's
-history to extract the events based on the updated configuration.
+Java Virtual Machine (JVM) Configuration
+----------------------------------------
 
+JVM configuration is important to ensure that PQS has enough resources
+to run efficiently. At minimum the following should be considered for
+any realistic deployment:
 
-PQS Development
-***************
+.. code:: text
 
-Offset Management for Querying
-==============================
+   -XX:+AlwaysPreTouch
+   -XX:-UseAdaptiveSizePolicy
 
-The following functions control the temporal perspective of the ledger,
-considering how you wish to consider time as a scope for your queries.
-You may wish to:
+   # containers:
+   -XX:InitialRAMPercentage=75.0
+   -XX:MaxRAMPercentage=75.0
 
--  Effectively ignore time; simply query the *latest available* state
--  Query the state of the ledger at a specific time in history
--  Query the ledger events across a time range -- an audit trail, for example
--  Query the ledger in a way that preserves consistency with other
-   interactions you have had with the ledger (reader or writer)
+   # host/vm:
+   -Xms4g
+   -Xmx4g
 
-The following functions allow you to control the temporal scope of the
-ledger, which establishes the context in which subsequent queries in the
-PostgreSQL session will execute:
+You may want to consider collecting heap-dumps if an out-of-memory error
```

Comment:
How about ... "In some situations an out-of-memory error may occur.  To diagnose this, you will need to collect heap-dumps."